lib/constructs/langfuse-ecr-fargate-deployment-cdk-stack.ts

```diff
@@ -0,0 +1,202 @@
+            },
+            loadBalancer,
+            securityGroups: [ecsSecurityGroup],
+            cpu: 1024,
+            memoryLimitMiB: 2048,
+            desiredCount: 1,
+            platformVersion: ecs.FargatePlatformVersion.LATEST,
+            runtimePlatform: {
+                cpuArchitecture: props.platformString === `arm` ? ecs.CpuArchitecture.ARM64 : ecs.CpuArchitecture.X86_64,
+                operatingSystemFamily: ecs.OperatingSystemFamily.LINUX,
+            },
+        });
+
+        // Setup AutoScaling policy
+        const scaling = fargateService.service.autoScaleTaskCount({ maxCapacity: 2, minCapacity: 1 });
+        scaling.scaleOnCpuUtilization(`${props.appName}-${props.environment}-${props.platformString}-CpuScaling`, {
+            targetUtilizationPercent: 70,
+            scaleInCooldown: cdk.Duration.seconds(60),
+            scaleOutCooldown: cdk.Duration.seconds(60)
+        });
+
+        fargateService.targetGroup.configureHealthCheck({
+            path: "/",
+            interval: cdk.Duration.seconds(60),
+            healthyHttpCodes: "200-499", // We have to check for 401 as the default state of "/" is unauthenticated
+        });
+
+        // Cloudfront Distribution
+        const langfuseOriginRequestPolicy = new OriginRequestPolicy(
+            this,
+            `${props.appName}-${props.environment}-${props.platformString}-OriginRequestPolicy`,
+            {
+                originRequestPolicyName: "LangfusePolicy",
+                comment: "Policy optimised for Langfuse",
+                cookieBehavior: OriginRequestCookieBehavior.all(),
+                headerBehavior: OriginRequestHeaderBehavior.all(),
+                queryStringBehavior: OriginRequestQueryStringBehavior.all(),
+            }
+        );
+
+        /** Fixes Cors Issue */
+        const corsFunction = new Function(this, `${props.appName}-${props.environment}-${props.platformString}-CorsFunction`, {
+            code: FunctionCode.fromInline(`
+                function handler(event) {
+                    if(event.request.method === 'OPTIONS') {
+                        var response = {
+                            statusCode: 204,
+                            statusDescription: 'OK',
+                            headers: {
+                                'access-control-allow-origin': { value: '*' },
+                                'access-control-allow-headers': { value: '*' }
+                            }
+                        };
+                        return response;
+                    }
+                    return event.request;
+                }
+            `),
+        });
+
+        const langfuseDistribution = new Distribution(this, `${props.appName}-${props.environment}-${props.platformString}-LangfuseDistribution`, {
+            defaultBehavior: {
+                origin: new LoadBalancerV2Origin(loadBalancer, {
+                    protocolPolicy: OriginProtocolPolicy.HTTP_ONLY,
```

Comment:
**Suggestion:** Change the `protocolPolicy` for the `LoadBalancerV2Origin` to `HTTPS_ONLY` to ensure secure communication between CloudFront and the load balancer.
> 🏰  **High Priority** 
 > [category: **`security`**, importance: **`9`**]
> **Reason:** Changing the protocol policy to `HTTPS_ONLY` is a critical security enhancement for protecting data in transit between CloudFront and the load balancer.
 

```suggestion
                    protocolPolicy: OriginProtocolPolicy.HTTPS_ONLY,
```